### PR TITLE
Updates for Boost 1.75 and TBB 2021.1.1

### PR DESCRIPTION
**Pull request by knoepfel**
These changes are based off of tag 0.14.0.  If you would like me to rebase them off of master, I can probably do that.

LArSoft would probably benefit from having a new release off of master--just let us know what versions of Jsonnet (etc.) are required.

### CODE REVIEW

**Comment by brettviren** [comment]
This is great.  Thanks, @knoepfel for figuring this out.  The fixes look good to me.  @HaiwangYu please also look them over.

Kyle, do you know what past versions of TBB support this `input_node` form?  Do we need to either specify a min version for TBB or try to include backwards compatibility?

BTW, I reference #103 here which has link also to the original redmine issue.

**Comment by HaiwangYu** [comment]
@knoepfel thanks for figuring this out! I will check this. And yes, we should request a LArSoft release soon for the master branch.

**Comment by knoepfel** [comment]
No problem, @brettviren.  Looks like TBB/oneAPI implemented the break (one among many) starting with version 2021.1.  See the [release notes](https://software.intel.com/content/www/us/en/develop/articles/intel-oneapi-threading-building-blocks-release-notes.html), in particular:

> New in 2021.1 Release
>
> **Changes affecting backward compatibility**
>
> - The code base was revamped to improve the usability and simplify the library, see [TBB Revamp: Background, Changes, and Modernization](https://software.intel.com/content/www/us/en/develop/articles/tbb-revamp.html). This version of the library is not backward compatible with any of the previous releases.

**Comment by brettviren** [comment]
Okay, thanks.  I think the "clients" that would be affected by requiring TBB 2021.1 for future builds are:

* FNAL UPS-land: here I think we just forge ahead with 2021.1 since it is this area that is driving the need for the TBB upgrade.
* CCE/PPS container: here it's your container, Kyle!  So, I guess it's okay to upgrade TBB there?
* The WCT developers.  It will likely mean some new dev environment builds.  I'll ping the mailing list to see if this causes any issues.

File: tbb/src/DataFlowGraph.cxx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-
-
 #include "WireCellTbb/DataFlowGraph.h"
 
 #include "WireCellUtil/Type.h"
 #include "WireCellUtil/NamedFactory.h"
 
+#include <tbb/task_scheduler_init.h>
```
Comment (brettviren) [inline tbb/src/DataFlowGraph.cxx:6]: This header seems to be gone from TBB 2021.1.1.  @knoepfel I wonder how you were able to build.  Maybe a second TBB include was in the build environment?

The "tbbrevamp" doc says `tbb::task_arena` or `tbb::global_control` are the replacement.  You have added the latter so we are in the right direction.  Is it clear how to make use of this replacement?

Comment (brettviren) [reply tbb/src/DataFlowGraph.cxx:6]: Looking through TBB includes and source, I'm not finding something equivalent to `automatic`.  I'm not even sure what that translated to.  I probably copied from some random example.

I'm tempted to make the default be simply `1`.  Any worries?

**Comment by brettviren** [comment]
Defaulting from the now missing `automatic` to `1` seems viable and with that I can remove the missing header and build locally against the new TBB and Boost.  I'll merge this PR and then follow up with a fix to remove the missing header `#include`.

**Pull request merged**